packages/terra-date-time-picker/src/DateTimeUtils.js

```diff
@@ -87,12 +87,16 @@ class DateTimeUtils {
     const newDate = momentDate.clone();
     const timeFormat = hasSeconds ? 'HH:mm:ss' : 'HH:mm';
     const date = moment(time, timeFormat, true);
+    const timeZone = momentDate.tz() || moment.tz.guess();
 
-    if (hasSeconds) {
-      return newDate.hour(date.get('hour')).minute(date.get('minute')).second(date.get('second'));
+    let dateTimeString;
+    if (date.isValid()) {
+      dateTimeString = newDate.format('YYYY-MM-DD').concat(' ').concat(date.format(timeFormat));
```

Comment:
not a big deal, but we could move the return statement inside the if block.
```suggestion
      dateTimeString = newDate.format('YYYY-MM-DD').concat(' ').concat(date.format(timeFormat));
      return moment.tz(dateTimeString, timeZone);
```

Reply:
updated here 1e47747132f83358d17d2aaed1f98c6bd4a5cfaf